lib/policial/limits_checker.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require 'logger'
+
+module Policial
+  # Public: Checks number of files received from GitHub and notifies if user
+  # reached GitHub per_page limit.
+  class LimitsChecker
+    DEFAULT_PERPAGE = 30
+
+    def initialize(github_client:, files:)
+      @github_client = github_client
+      @files = files
+    end
+
+    def check
+      return if @github_client.auto_paginate
+      return if (@github_client.per_page || DEFAULT_PERPAGE) > @files.count
```

Comment:
Yeah, I don't think counting items like this is very safe. GitHub provides a way to check if the response was paginated by the presence of the `Link` header. In Octokit, that can be checked via `client.last_response.rels`.

What about doing that instead of comparing numbers? Something like

```ruby
return if @github_client.auto_paginate
return if @github_client.last_response.rels.empty?
raise ...
```